storage/google/cloud/storage/blob.py

```diff
@@ -1530,25 +1532,15 @@ def update_storage_class(self, new_class, client=None):
         if new_class not in self._STORAGE_CLASSES:
             raise ValueError("Invalid storage class: %s" % (new_class,))
 
-        client = self._require_client(client)
-
-        query_params = {}
-
-        if self.user_project is not None:
-            query_params["userProject"] = self.user_project
-
-        headers = _get_encryption_headers(self._encryption_key)
-        headers.update(_get_encryption_headers(self._encryption_key, source=True))
+        # Update current blob's storage class prior to rewrite
+        self._patch_property('storageClass', new_class)
 
-        api_response = client._connection.api_request(
-            method="POST",
-            path=self.path + "/rewriteTo" + self.path,
-            query_params=query_params,
-            data={"storageClass": new_class},
-            headers=headers,
-            _target_object=self,
-        )
-        self._set_properties(api_response["resource"])
+        # Execute consecutive rewrite operations until operation is done
+        token = None
+        while True:
+            token, _, _ = self.rewrite(self, token=token)
+            if token is None:    # pragma: no branch
```

Comment:
I added `# pragma: no branch` to bypass code coverage check on both branches. In order to implement both branches, I need to receive 2 different, subsequent responses from `rewrite()` above.

Reply:
https://coverage.readthedocs.io/en/v4.5.x/branch.html#structurally-partial-branches

Reply:
@erikwebb We normally set up separate testcases for that kind of issue.

Reply:
@tseaver Do you have a pointer on other mocks that return subsequent results like I need here? I.e. first response from `self.rewrite()` includes `token is not None`, while the second call returns `token is None`.

Reply:
@erikwebb You can set the `side_effect` attribute of a mocked callable to a list of items, which will be returned in order across multiple calls.  E.g., see [`_mock_download_transport`](https://github.com/googleapis/google-cloud-python/blob/523b7b70ab3fdc85b0dbefb5f42d0bdd7ffcd723/storage/tests/unit/test_blob.py#L696-L710)

Reply:
Thanks! By looking through that, I finally was led to `class _Connection` supporting this natively.